packages/test/test-end-to-end-tests/src/test/nodeDidNotRunGc.spec.ts

```diff
@@ -0,0 +1,277 @@
+export const TestDataObjectType2 = "@fluid-example/test-dataStore2";
+class TestDataObject2 extends DataObject {
+	public get _root() {
+		return this.root;
+	}
+	public get _context() {
+		return this.context;
+	}
+}
+class TestDataObject1 extends DataObject {
+	public get _root() {
+		return this.root;
+	}
+
+	public get _context() {
+		return this.context;
+	}
+
+	protected async initializingFirstTime() {}
+
+	protected async hasInitialized() {
+		const dsFactory2 = await requestFluidObject<TestDataObject2>(
+			await this._context.containerRuntime.createDataStore(TestDataObjectType2),
+			"",
+		);
+		assert(dsFactory2 !== undefined);
+		Promise.resolve()
+			.then(() => {
+				this.root.set("dsFactory2", dsFactory2.handle);
+			})
+			.catch(console.error);
+	}
+}
+
+describeNoCompat("NodeDidNotRunGC", (getTestObjectProvider) => {
+	let provider: ITestObjectProvider;
+	const dataStoreFactory1 = new DataObjectFactory(
+		TestDataObjectType1,
+		TestDataObject1,
+		[SharedMatrix.getFactory(), SharedCounter.getFactory()],
+		[],
+		[],
+		createDataStoreRuntime(),
+	);
+	const dataStoreFactory2 = new DataObjectFactory(
+		TestDataObjectType2,
+		TestDataObject2,
+		[],
+		[],
+		[],
+		createDataStoreRuntime(),
+	);
+	const innerRequestHandler = async (request: IRequest, runtime: IContainerRuntimeBase) =>
+		runtime.IFluidHandleContext.resolveHandle(request);
+	const registryStoreEntries = new Map<string, Promise<IFluidDataStoreFactory>>([
+		[dataStoreFactory1.type, Promise.resolve(dataStoreFactory1)],
+		[dataStoreFactory2.type, Promise.resolve(dataStoreFactory2)],
+	]);
+
+	const createContainer = async (
+		disableSummary: boolean = true,
+		logger?: ITelemetryLogger,
+	): Promise<IContainer> => {
+		let summaryConfigOverrides: ISummaryConfiguration;
+		if (disableSummary) {
+			summaryConfigOverrides = { state: "disabled" };
+		} else {
+			const IdleDetectionTime = 100;
+			summaryConfigOverrides = {
+				...DefaultSummaryConfiguration,
+				...{
+					maxIdleTime: IdleDetectionTime,
+					maxTime: IdleDetectionTime * 12,
+					initialSummarizerDelayMs: 0,
+				},
+			};
+		}
+		const runtimeOptions: IContainerRuntimeOptions = {
+			summaryOptions: {
+				summaryConfigOverrides,
+			},
+		};
+		const runtimeFactory = new ContainerRuntimeFactoryWithDefaultDataStore(
+			dataStoreFactory1,
+			registryStoreEntries,
+			undefined,
+			[innerRequestHandler],
+			runtimeOptions,
+		);
+		return provider.createContainer(runtimeFactory, { logger });
+	};
+
+	async function waitForSummaryOp(container: IContainer) {
+		await new Promise<void>((resolve) => {
+			container.deltaManager.on("op", (op: ISequencedDocumentMessage) => {
+				if (op.type === MessageType.Summarize) {
+					resolve();
+				}
+			});
+		});
+	}
+
+	beforeEach(async () => {
+		provider = getTestObjectProvider({ syncSummarizer: true });
+	});
+
+	itExpects(
+		"summarizeOnDemand should fail with NodeDidNotRunGC error",
```

Comment:
The point is (in terms of the difference between the tests) that on demand doesn't have the retry loop?  Do we want to put the retry behind a feature gate?  Hopefully don't feel the need to but we should think about it.

Reply:
@agarwal-navin - Do we want to put the retry bit behind a feature gate?

Reply:
We can put disabling it behind a feature gate to be safe.

Reply:
I will make a change to add a feature flag to be safe.